src/en/mangamonks/src/eu/kanade/tachiyomi/extension/en/mangamonks/MangaMonks.kt

```diff
@@ -0,0 +1,267 @@
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    // search
+    private var searchMode: Boolean = false
+
+    private val searchUrl = "https://mangamonks.com/search/live"
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val filterList = filters.let { if (it.isEmpty()) getFilterList() else it }
+        return if (query.isNotEmpty()) {
+            searchMode = true
+            val requestBody = query.toFormRequestBody()
+            val requestHeaders = headersBuilder().buildApiHeaders(requestBody)
+
+            POST(searchUrl, requestHeaders, requestBody)
+        } else {
+            searchMode = false
+            val url = "$baseUrl/genre/".toHttpUrl().newBuilder()
+            filterList.forEach { filter ->
+                when (filter) {
+                    is GenreFilter -> filter.toUriPart().let {
+                        url.apply {
+                            addPathSegment(it)
+                            addQueryParameter("include[]", filter.toGenreValue())
+                        }
+                    }
+                    is StatusFilter -> filter.toUriPart().let {
+                        url.apply {
+                            addQueryParameter("term", query)
+                            addQueryParameter("status[]", it)
+                        }
+                    }
+                    else -> {}
+                }
+            }
+
+            url.addPathSegment(page.toString())
+            GET(url.build(), headers)
+        }
+    }
+
+    override fun searchMangaSelector() = ".main-slide .item"
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    private val json: Json by injectLazy()
+    override fun searchMangaParse(response: Response): MangasPage {
+        try {
+            if (searchMode) {
+                val result = json.decodeFromString<MangaList>(response.body.string())
+                val mangaList = result.manga.map {
+                    SManga.create().apply {
+                        title = it.title
+                        setUrlWithoutDomain(it.url)
+                        thumbnail_url = it.image
+                    }
+                }
+                val hasNextPage = false
+                return MangasPage(mangaList, hasNextPage)
+            } else {
+                val document = response.asJsoup()
+
+                val mangas = document.select(searchMangaSelector()).map { element ->
+                    searchMangaFromElement(element)
+                }
+
+                val hasNextPage = searchMangaNextPageSelector().let { selector ->
+                    document.select(selector).first()
+                } != null
+
+                return MangasPage(mangas, hasNextPage)
+            }
+        } catch (_: MissingFieldException) {
```

Comment:
This try/catch should (if anything) be wrapped around `json.decodeFromString`, not the whole function body

Reply:
This try is necessary because the site doesn't accept search queries that are less than 3 chars, I wrapped it around the json.decodeFromString section